### PR TITLE
Feat/aa/pagination controls layout

### PR DESCRIPTION
**Pull request by andres-acc**
## Summary
* Add paginator component

ezgif com-video-to-gif (2)

## [Trello Card ](https://trello.com/c/Lqnsm4is)

### CODE REVIEW

File: src/app/components/paginator/paginator.component.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+          ...pagesArray.splice(0, 2), 
+          -1, 
+          ...pagesArray.splice(pagesArray.length - 2, pagesArray.length)
+        ];
+        this.pagesToShow = pagesArray;
+      } else if(pagesArray.length === 4) {
+        pagesArray = [
+          -1, 
+          ...pagesArray.splice(0, 2), 
+          ...pagesArray.splice(pagesArray.length - 2, pagesArray.length)
```
Comment (joacomiguens) [inline src/app/components/paginator/paginator.component.ts:45]: maybe you could assign this to a const and use that const

File: src/app/components/paginator/paginator.component.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    }
+  }
+
+  goToPage(page: number): void {
+    if(!page) return;
+    this.gotToPageEvent.emit(page);
+    this.definePagesButtons(page);
+  }
+
+  goToPageFromArrow(arrow: 'left' | 'right'): void {
```
Comment (joacomiguens) [inline src/app/components/paginator/paginator.component.ts:58]:
```suggestion
  goToPageFromArrow(go: 'back' | 'foward'): void {
```

File: src/app/components/paginator/page-button/page-button.component.scss [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    border-color: $wolox-blue;
+  }
+}
+
+.left {
+  &::after {
+    @include arrow;
+    transform: rotate(225deg);
+  }
+}
```
Comment (miguelhenaoa) [inline src/app/components/paginator/page-button/page-button.component.scss:33]: I think that we can refactor this code 
```suggestion
.left::after {
  @include arrow;
  transform: rotate(225deg);
}
```

File: src/app/components/paginator/page-button/page-button.component.scss [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+.right {
+  &::after {
+    @include arrow;
+    transform: rotate(45deg);
+  }
+}
```
Comment (miguelhenaoa) [inline src/app/components/paginator/page-button/page-button.component.scss:40]: Same here

File: src/app/components/paginator/page-button/page-button.component.scss [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+  border-color: $wolox-blue;
+  color: $wolox-blue;
+}
+
+.dots {
+  &::after {
+    content: '...';
+    letter-spacing: 2px;
+  }
+}
```
Comment (miguelhenaoa) [inline src/app/components/paginator/page-button/page-button.component.scss:62]: Same here

File: src/app/components/paginator/paginator.component.html [hunk 6]
```diff
@@ -0,0 +1,18 @@
+<app-page-button 
+  [leftArrow]="true"
+  [disabled]="currentPage === 1"
+  (clickEvent)="goToPageFromArrow('left')"
+></app-page-button>
+<ng-container *ngFor="let page of pagesToShow">
+  <app-page-button 
+    [number]="page !== -1 ? page : undefined"
+    [dots]="page === -1"
+    [current]="page === currentPage"
```
Comment (miguelhenaoa) [inline src/app/components/paginator/paginator.component.html:10]: Maybe we can move this validations to TS file

File: src/app/components/paginator/paginator.component.ts [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+  selector: 'app-paginator',
+  templateUrl: './paginator.component.html',
+  styleUrls: ['./paginator.component.scss']
+})
+export class PaginatorComponent implements OnInit {
+  @Input() pages: number = 0;
+  @Input() currentPage: number = 0;
+  @Input() visiblePages: number = 0;
+  
+  @Output() gotToPageEvent = new EventEmitter<number>();
```
Comment (miguelhenaoa) [inline src/app/components/paginator/paginator.component.ts:19]: Fix typo `goToPageEvent`

File: src/app/components/paginator/paginator.component.ts [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+  }
+
+  goToPageFromArrow(go: 'left' | 'right'): void {
+    if(go === 'left') {
+      this.gotToPageEvent.emit(this.currentPage - 1);
+      this.definePagesButtons(this.currentPage - 1);
+    } else {
+      this.gotToPageEvent.emit(this.currentPage + 1);
+      this.definePagesButtons(this.currentPage + 1);
+    }
```
Comment (miguelhenaoa) [inline src/app/components/paginator/paginator.component.ts:70]: We can refactor this code
```suggestion
    const newPage = go === 'left' ? this.currentPage - 1 : this.currentPage + 1;
    this.gotToPageEvent.emit(newPage);
    this.definePagesButtons(newPage);
```

**Review by miguelhenaoa — CHANGES_REQUESTED** [review]

**Review by miguelhenaoa — APPROVED** [review]
LGTM!

**Pull request merged**